packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -721,7 +726,7 @@ function validateStyleAttr(
     return nodesAreCompatible;
 }
 
-function areCompatibleNodes(client: Node, ssr: Node, vnode: VNode, renderer: RendererAPI) {
+function areCompatibleNodes(client: Node, ssr: Node, vnode: VStatic, renderer: RendererAPI) {
```

Comment:
```suggestion
function areCompatibleStaticNodes(client: Node, ssr: Node, vnode: VStatic, renderer: RendererAPI) {
```

Otherwise it's not clear why this is only scoped to VStatic.